src/module.c

```diff
@@ -74,6 +138,30 @@ static inline bool SearchCluster_Ready() {
   return NumShards != 0;
 }
 
+static bool ACLUserMayAccessIndex(RedisModuleCtx *ctx, IndexSpec *sp) {
+  if (RedisModule_ACLCheckKeyPrefixPermissions == NULL) {
+    // Running against a Redis version that does not support module ACL protection
+    RedisModule_Log(ctx, "warning", "Redis version does not support ACL API necessary for index protection");
+    return true;
+  }
+  RedisModuleString *user_name = RedisModule_GetCurrentUserName(ctx);
+  RedisModuleUser *user = RedisModule_GetModuleUserFromUserName(user_name);
+
+  if (!user) {
+    RedisModule_Log(ctx, "warning", "No user found");
```

Comment:
It shouldn't happen as well, and notice that we don't allow the command to be executed in this case, so I think it's quite critical.

Reply:
This goes to the log, if this went back as a reply then fine.
Maybe at least log the command that the user tried running?

In the documentation it says it can happen if the user is disabled.
Isn't it the same as to say the user doesn't have permissions?

Btw, aren't you missing a free call in the case of return false?

Reply:
Maybe get the `RedisModuleUser*` as an argument?.
Then you can return an error in the macro.

Reply:
Changed to `verbose`